### ISSUE TITLE
Prop type error every time I select a new day

### ISSUE THREAD

**Issue by SandroMachado** [clauderic/react-infinite-calendar#60]
The calendar component renders as expected, the problem is every time I select a new day, I get a lot of errors in the console.

My render function:
```
  render() {
    var today = new Date();
    var minDate = Number(new Date()) - (24 * 60 * 60 * 1000) * 7;

    return (
      <InfiniteCalendar
        selectedDate={today}
        minDate={minDate}
        keyboardSupport={true}
        onSelect={ date => {
          console.log("DATE", date);
        }}
      />
    );
  }
```

Every time I click in a new day I get a lot of errors in the console:

```
warning.js:36 Warning: Failed prop type: Invalid prop `shouldHeaderAnimate` of type `object` supplied to `Header`, expected `boolean`
...
warning.js:36 Warning: Failed prop type: Invalid prop `transitionEnter` of type `object` supplied to `ReactCSSTransitionGroup`, expected `boolean`
...
warning.js:36 Warning: Failed prop type: Invalid prop `transitionLeave` of type `object` supplied to `ReactCSSTransitionGroup`, expected `boolean`.
...
warning.js:36 Warning: Failed prop type: Invalid prop `enter` of type `object` supplied to `ReactCSSTransitionGroupChild`, expected `boolean`.
    in ReactCSSTransitionGroupChild (created by ReactTransitionGroup)
...
warning.js:36 Warning: Failed prop type: Invalid prop `leave` of type `object` supplied to `ReactCSSTransitionGroupChild`, expected `boolean`.
    in ReactCSSTransitionGroupChild (created by ReactTransitionGroup)
```

Any thoughts how to fix this?

**Comment by clauderic** [comment]
Hmm, which version of react / react-addons-css-transition-group do you have installed locally?

**Comment by SandroMachado** [comment]
Sure:

`react` - **15.3.2**
`react-addons-css-transition-group` - **15.4.1**

Thanks for the reply.

**Comment by Flamenco** [comment]
I get this too.  The headerShouldAnimate property is the click event instead of the boolean value.

**Comment by Flamenco** [comment]
`_this.onDaySelect = function (selectedDate, e) {
			var shouldHeaderAnimate = arguments.length <= 2 || arguments[2] === undefined ? _this.props.shouldHeaderAnimate : arguments[2];`

The problem is arguments size is 3. [Moment, Proxy, Event].

**Comment by Flamenco** [comment]
In index.js I changed 

`var shouldHeaderAnimate = arguments.length <= 3 || arguments[3] === undefined ? _this.props.shouldHeaderAnimate : arguments[3];`

I'm not really sure where the piggy-backed expected value would be coming from though.

**Comment by SandroMachado** [comment]
@Flamenco yup, it works. Changing from `2` to `3` removes the errors. 👍 

@clauderic Do you have any hints why the distribution files are getting generated with `2` instead of `3`?

**Comment by Flamenco** [comment]
That's the babel transpiler for an ES6 default option.  The 'proxy' is not expected in the function call.  The function is expecting [Moment, Event, shouldHeaderAnimate {optional} ].  

So the '2' is correct. Something is generating an extra, unexpected argument.

**Comment by Flamenco** [comment]
<img width="677" alt="screen shot 2016-12-12 at 1 50 23 pm" src="https://cloud.githubusercontent.com/assets/819940/21112014/fd7ffb76-c071-11e6-93e4-a6e442de9ed3.png">

**Comment by xtalx** [comment]
I get this as well (the same 5 errors listed by the OP).
react-addons-css-transition-group@15.4.2
react-dom@15.4.2
react-infinite-calendar@1.2.2

I noticed that the errors are only generated the **first** time the the InfiniteCalendar component is mounted.

**Comment by embengineering** [comment]
+1 I've experience the same, only after the component is mounted.  Then it works as expected.

**Comment by leafi** [comment]
Hi, please see https://github.com/toggl/react-infinite-calendar/commit/fd1740ca0d8a9bd871e0b10ca6e58f2ec020a1e0 for one possible fix and a little extra commentary.

In effect, one of the things React does related to the creation of synthetic events is it passes more than one parameter to e.g. the onClick event handler.

While the 'second' argument was often undefined in one specific version of React, that changed recently. This was triggering the third, optional argument passed to onDaySelect to (sometimes?) be neither null nor undefined, causing a random event-like object to be copied to shouldHeaderAnimate in the component's state - causing the failed prop type/invalid prop error we're experiencing here.

Thanks for the messages above - they were extremely helpful in figuring out the issue.

**Comment by clauderic** [comment]
Hey guys, just shipped version `2.0.0`, which comes with loads of new features and should also fix this issue. Give it a try and let me know if you run into any issues!

Check out #73 for more details 😄
